passive/Server Header Disclosure.js

```diff
@@ -14,10 +15,38 @@ function scan(ps, msg, src)
 
     var url = msg.getRequestHeader().getURI().toString();
     var headers = msg.getResponseHeader().getHeaders("Server")
+    var headers_string = headers.toString();
     
-    if (headers != null)
+    if (headers != null && ExtAlert(headers))
     {
-        ps.raiseAlert(alertRisk, alertConfidence, alertTitle, alertDesc, url, '', '', '', alertSolution,headers, cweId, wascId, msg);
+        ps.raiseAlert(alertRisk, alertConfidence, alertTitle, alertDesc, url, '', '', '', alertSolution, headers_string, cweId, wascId, msg);
     }
     
 }
+
+function ExtAlert(content) {
+
+    var ext = new RegExp("(\\d+\\.)+\\d+");
```

Comment:
Why does this have a capturing group if the potential matches aren't returned?

You could make it non-capturing with `?:` .
https://www.stefanjudis.com/today-i-learned/non-capturing-groups-in-regular-expressions/

Reply:
used non-capture group.